R/expect-condition.R

```diff
@@ -301,6 +297,13 @@ cnd_matcher <- function(base_class,
   check_string(class, allow_null = TRUE, call = error_call)
   check_string(pattern, allow_null = TRUE, allow_na = TRUE, call = error_call)
 
+  if (is.null(pattern) && dots_n(...) > 0) {
+    cli::cli_abort(
```

Comment:
It looks like we warned before, just confirming you want an error now?

Before:

```r
 warn(conditionMessage(cnd), call = error_call)
```

Reply:
Right, because I'm more convinced now that this is a real error.

Reply:
I _think_ previously I thought that `grepl()` accepted `...`? I'm not sure.

Reply:
🤔 But you don't ever call `grepl()` when `pattern` is NULL right? And that's what this is saying, I think. Just that those `...` aren't going to be used

Reply:
I think I agree it should probably be an error, I would just be more aggressive in the message

```
"{.arg ...} ignored when {.arg pattern} is not set."
```

sounds like a warning but maybe:

```
"Can't specify {.arg ...} when {.arg pattern} is not set."
```

is better for an error?

Reply:
Oooh got it.